OMEdit/OMEditGUI/Animation/ViewerWidget.cpp

```diff
@@ -248,21 +249,81 @@ void ViewerWidget::showShapePickContextMenu(const QPoint& pos)
+  QAction action4(QIcon(":/Resources/icons/checkered.svg"), tr("Apply Check Texture"), this);
+  QAction action5(QIcon(":/Resources/icons/texture.svg"), tr("Apply Custom Texture"), this);
 
   //if a shape is picked, we can set it transparent
   if (0 != QString::compare(name,QString(""))) {
     contextMenu.addMenu(&shapeMenu);
     shapeMenu.addAction( &action1);
     shapeMenu.addAction( &action2);
+    shapeMenu.addAction( &action3);
+    shapeMenu.addAction( &action4);
+    shapeMenu.addAction( &action5);
     connect(&action1, SIGNAL(triggered()), this, SLOT(changeShapeTransparency()));
     connect(&action2, SIGNAL(triggered()), this, SLOT(makeShapeInvisible()));
+    connect(&action3, SIGNAL(triggered()), this, SLOT(changeShapeColor()));
+    connect(&action4, SIGNAL(triggered()), this, SLOT(applyCheckTexture()));
+    connect(&action5, SIGNAL(triggered()), this, SLOT(applyCustomTexture()));
+
   }
   contextMenu.addAction(&action0);
   connect(&action0, SIGNAL(triggered()), this, SLOT(removeTransparencyForAllShapes()));
   contextMenu.exec(this->mapToGlobal(pos));
 }
 
 /*!
+ * \brief ViewerWidget::applyCheckTexture
+ * changes the transparency selection of a shape
+ */
+void ViewerWidget::applyCheckTexture()
+{
+    ShapeObject* shape = nullptr;
+    if ((shape = mpAnimationWidget->getVisualizer()->getBaseData()->getShapeObjectByID(mSelectedShape)))
+    {
+      if (shape->_type.compare("dxf") == 0 or shape->_type.compare("stl") == 0)
+      {
+        QString msg = tr("Texture feature for CAD-Files is not applicable.");
+        MessagesWidget::instance()->addGUIMessage(MessageItem(MessageItem::Modelica, "", false, 0, 0, 0, 0, msg, Helper::scriptingKind,
+                                                                    Helper::notificationLevel));
+      }
+      else
+      {
+        shape->setTextureImagePath(":/Resources/bitmaps/check.png");
+        mpAnimationWidget->getVisualizer()->modifyShape(mSelectedShape);
+        mSelectedShape = "";
+      }
+    }
+}
+
+/*!
+ * \brief ViewerWidget::applyCustomTexture
+ * changes the transparency selection of a shape
+ */
+void ViewerWidget::applyCustomTexture()
+{
+    ShapeObject* shape = nullptr;
+    if ((shape = mpAnimationWidget->getVisualizer()->getBaseData()->getShapeObjectByID(mSelectedShape)))
+    {
+      if (shape->_type.compare("dxf") == 0 or shape->_type.compare("stl") == 0)
+      {
+        QString msg = tr("Texture feature for CAD-Files is not applicable.");
+        MessagesWidget::instance()->addGUIMessage(MessageItem(MessageItem::Modelica, "", false, 0, 0, 0, 0, msg, Helper::scriptingKind,
+                                                                    Helper::notificationLevel));
+      }
+      else
+      {
+        QString fileName = StringHandler::getOpenFileName(this, QString("%1 - %2").arg(Helper::applicationName).arg(Helper::chooseFile),
+                                                              NULL, Helper::bitmapFileTypes, NULL);
```

Comment:
You need to handle the case when user press Esc. In that case fileName would be empty.